File: src/app/home/Song.ts
import {ISong} from "../../plugin/song-finder";

export class Song {
    public readonly album;
    public readonly artist;
    public readonly name;
    public readonly length;
    public readonly modificationTime;
    public readonly mimetype;
    public readonly nativeURL;
    public readonly genre;

    public constructor(song: ISong, public bpm: number) {
        this.album = song.album;
        this.artist = song.artist;
        this.name = song.title;
        this.length = song.length / 1000;
        this.modificationTime = song.modificationTime;
        this.mimetype = song.mimetype;
        this.nativeURL = song.nativeURL;
        this.genre = song.genre || [];
    }
}

File: android/app/src/main/java/plugins/SongFinder/Song.java
package plugins.SongFinder;

import java.time.LocalDate;

public class Song {
    private final String nativeURL;
    private final int modificationTime;
    private final String mimetype;
    private final String artist;
    private final String album;
    private final String title;
    private final long length;

    public Song(String nativeURL, int modificationTime, String mimetype, String artist, String album, String title, long length) {
        this.nativeURL = nativeURL;
        this.modificationTime = modificationTime;
        this.mimetype = mimetype;
        this.artist = artist;
        this.album = album;
        this.title = title;
        this.length = length;
    }

    public String getNativeURL() {
        return nativeURL;
    }

    public int getModificationTime() {
        return modificationTime;
    }

    public String getMimetype() {
        return mimetype;
    }

    public String getArtist() {
        return artist;
    }

    public String getAlbum() {
        return album;
    }

    public String getTitle() {
        return title;
    }

    public long getLength() {
        return length;
    }
}
